core/src/main/java/hudson/model/Descriptor.java

```diff
@@ -807,8 +809,33 @@ public boolean configure( StaplerRequest req ) throws FormException {
      *      to keep the client in the same config page.
      */
     public boolean configure( StaplerRequest req, JSONObject json ) throws FormException {
-        // compatibility
-        return configure(req);
+
+        if (Util.isOverridden(Descriptor.class, getClass(), "configure", StaplerRequest.class)) {
+            // compatibility
+            return configure(req);
+        }
+
+        try (BulkChange bc = new BulkChange(this)) {
+            req.bindJSON(this, json);
+            bc.commit();
+        } catch (IOException e) {
+            LOGGER.log(Level.WARNING, "Failed to save "+getConfigFile(),e);
```

Comment:
I'm a bit worried we hide failure to save configuration here. Should throw a FormException. wdyt ?

Reply:
Agree, but FormException expect a specific field, and here the whole binding process did fail. Maybe better to throw an IOException ?
No concrete sample I can find in code to see the conventional way to address this

Reply:
Actually (sorry, this PR being 1 month old I can't remember such details) the original code forhudson.model.Descriptor#save doesn't send any feedback to end-user but just log failure to persist configuration, so I just did the same here.